### PR TITLE
Refactoring: moving find_string to libutil

### PR DESCRIPTION
**Pull request by agrawalravi90**
#### Bug/feature Description
* Many interesting string/string array related functions are already part of libutil, find_string() is a scheduler function that is generic and useful for string arrays, so it makes sense to move it to libutil.

#### Affected Platform(s)
* All

#### Cause / Analysis / Design
* Not much here, just some refactoring which will help other parts of PBS use these functions.

#### Solution Description
* Moved find_string() and find_string_ind() to libutil. Also renamed find_string to "is_string_in_arr" and find_string_ind to "find_string_idx"

#### Testing logs/output
* Since this is just refactoring, it would be sufficient if the smoke tests pass.

#### Checklist:

- [X] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [X] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [X] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
   - [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
   - [ ] I have added  **manual test(s) to this pull request and explained why PTL is not appropriate** for this case.
- [X] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [ ] I have attached **test logs to this pull request** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

**Comment by SudeshnaMoulik** [comment]
Changes looks good to me.

**Comment by agrawalravi90** [comment]
Thanks @SudeshnaMoulik ! @PBSPro/pbspro-maintainers requesting a maintainer review, very simple change, thanks!

File: src/include/libutil.h [hunk 1]
```diff
@@ -257,6 +257,16 @@ char **break_comma_list(char *list);
...
 
+/*
+ * find index of str in strarr
+ */
+int find_string_ind(char **strarr, char *str);
+
+/*
+ *	find_string - find a string in a NULL terminated string array
+ */
+int find_string(char **strarr, char *str);
```
Comment (subhasisb) [inline src/include/libutil.h]: The name find_string() sounds quite misleading (especially if it has to become a generically useful function to the rest of PBS). It sounds like it is a replacement for strstr(). However, it finds a string in an array of strings. Could we rename it to something better?

File: src/lib/Libutil/misc_utils.c [hunk 2]
```diff
@@ -1192,6 +1192,56 @@ break_comma_list(char *strlist)
...
+ *
+ * @param[in]	strarr	-	the string array to search
+ * @param[in]	str	-	the string to find
+ *
+ * @return	int
+ * @retval	index of string
+ * @retval	-1	: if not found
+ */
+int
+find_string_ind(char **strarr, char *str)
```
Comment (subhasisb) [inline src/lib/Libutil/misc_utils.c]: Is this a static function to be used only by find_string()

Comment (agrawalravi90) [reply src/lib/Libutil/misc_utils.c]: No, it is called directly as well by the scheduler

**Comment by agrawalravi90** [comment]
@subhasisb thanks for your comments, I've renamed the functions, please let me know if the new names make sense now.

**Pull request merged**